benches/benches/block_target_gas.rs

```diff
@@ -138,6 +154,209 @@ fn run(
+    let state_size_env_var = "STATE_SIZE";
+    let state_size = if let Some(value) = std::env::var_os(state_size_env_var) {
+        let value = value.to_str().unwrap();
+        let value = value.parse::<u64>().unwrap();
+        println!("Overriding state size with {}", value);
+        value
+    } else {
+        STATE_SIZE
+    };
+
+    database
+        .init_contract_state(
+            &contract_id,
+            (0..state_size).map(|k| {
+                let mut key = Bytes32::zeroed();
+                key.as_mut()[..8].copy_from_slice(&k.to_be_bytes());
+                (key, key)
+            }),
+        )
+        .unwrap();
+    database
+        .init_contract_balances(
+            &contract_id,
+            (0..state_size).map(|k| {
+                let key = k / 2;
+                let mut sub_id = Bytes32::zeroed();
+                sub_id.as_mut()[..8].copy_from_slice(&key.to_be_bytes());
+
+                let asset = if k % 2 == 0 {
+                    VmBench::CONTRACT.asset_id(&sub_id)
+                } else {
+                    AssetId::new(*sub_id)
+                };
+                (asset, key + 1_000)
+            }),
+        )
+        .unwrap();
+
+    let service = fuel_core::service::FuelService::new(database, config.clone())
+        .expect("Unable to start a FuelService");
+    service.start().expect("Unable to start the service");
+    (service, rt)
+}
+
+/// Runs benchmark for `script` with prepared `service` and specified contract (by 'contract_id') which should be
+/// included in service
+#[allow(clippy::too_many_arguments)]
+fn run_with_service(
+    id: &str,
+    group: &mut BenchmarkGroup<WallTime>,
+    script: Vec<Instruction>,
+    script_data: Vec<u8>,
+    service: &fuel_core::service::FuelService,
+    contract_id: ContractId,
+    rt: &tokio::runtime::Runtime,
+    rng: &mut rand::rngs::StdRng,
+) {
+    run_with_service_with_extra_inputs(
+        id,
+        group,
+        script,
+        script_data,
+        service,
+        contract_id,
+        rt,
+        rng,
+        vec![],
+        vec![],
+    );
+}
+
+/// Runs benchmark for `script` with prepared `service` and specified contract (by `contract_id`) which should be
+/// included in service.
+/// Also include additional inputs and outputs in transaction
+#[allow(clippy::too_many_arguments)]
+fn run_with_service_with_extra_inputs(
+    id: &str,
+    group: &mut BenchmarkGroup<WallTime>,
+    script: Vec<Instruction>,
+    script_data: Vec<u8>,
+    service: &fuel_core::service::FuelService,
+    contract_id: ContractId,
+    rt: &tokio::runtime::Runtime,
+    rng: &mut rand::rngs::StdRng,
+    extra_inputs: Vec<Input>,
+    extra_outputs: Vec<Output>,
+) {
+    group.bench_function(id, |b| {
+
+        b.to_async(rt).iter(|| {
+            let shared = service.shared.clone();
+
+
+            let mut tx_builder = fuel_core_types::fuel_tx::TransactionBuilder::script(
+                // Infinite loop
```

Comment:
The comment here seems out of place. Isn't this an infinite loop only if the passed `script` loops?

Reply:
Nice catch. It's a relic from older code.